sage-ios/SAGE/SAGE/AnnouncementsViewController.swift

```diff
@@ -121,9 +140,55 @@ class AnnouncementsViewController: UITableViewController {
         }
         self.navigationController?.pushViewController(addAnnouncementController, animated: true)
     }
-    
-    func getAnnouncements() {
-        AnnouncementsOperations.loadAnnouncements({ (announcements) -> Void in
+
+    func showFilterOptions() {
+        let menuController = MenuController(title: "Filter Options")
+        menuController.addMenuItem(MenuItem(title: "All", handler: { (_) -> Void in
+            self.filter = nil
+            self.getAnnouncements(reset: true)
+            self.titleView.setSubtitle("All")
+        }))
+
+        if LoginOperations.getUser()!.isDirector() {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(LoginOperations.getUser()!.directorID)]
+                self.getAnnouncements(reset: true)
+                self.titleView.setSubtitle("My School")
+            }))
+        } else if let userSchool = KeychainWrapper.objectForKey(KeychainConstants.kSchool) as? School {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(userSchool.id)]
+                self.getAnnouncements(reset: true)
+                self.titleView.setSubtitle(userSchool.name!)
+            }))
+        }
+
```

Comment:
there can be some corner cases for directors if they change their school, but we wanna prioritize the school they direct